Sources/Workspace/Workspace.swift

```diff
@@ -290,7 +290,7 @@ public class Workspace {
     ) throws {
         // defaults
         let currentToolsVersion = customToolsVersion ?? ToolsVersion.currentToolsVersion
-        let toolsVersionLoader = ToolsVersionLoader()
+        let toolsVersionLoader = ToolsVersionLoader(currentToolsVersion: currentToolsVersion)
```

Comment:
customToolsVersion should only be used in tests, we should probably  remove it from the public constructor and move it into an internal one

Reply:
addressed in https://github.com/apple/swift-package-manager/pull/3933